### PR TITLE
[21.09] Fix installation of tools with biotools id

### LINKED ISSUE #12918: Error when installing fastqc or anything with populated <xref type="bio.tools"> on 21.09
**Describe the bug**
I installed fastqc on an instance of galaxy running the release_21.09 branch and it reported very quickly that it had installed, but the tool did not show up in the tool panel and is not returned from bioblend galaxy.tools.get_tools().  The conda environment has not installed.  This error is in the log:

```
galaxy.tool_shed.tools.tool_validator ERROR 2021-11-15 15:55:52,992 [pN:main.web.1,p:5536,w:1,m:0,tN:uWSGIWorker1Core3] Caught exception loading tool from /Users/cat/dev/galaxy/database/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/3d0c7bdf12f5/fastqc/rgFastQC.xml:
Traceback (most recent call last):
  File "lib/galaxy/tool_shed/tools/tool_validator.py", line 80, in load_tool_from_config
    tool = create_tool_from_source(config_file=full_path, app=self.app, tool_source=tool_source, repository_id=repository_id, allow_code_files=False)
  File "lib/galaxy/tools/__init__.py", line 266, in create_tool_from_source
    tool = ToolClass(config_file, tool_source, app, **kwds)
  File "lib/galaxy/tools/__init__.py", line 585, in __init__
    raise e
  File "lib/galaxy/tools/__init__.py", line 582, in __init__
    self.parse(tool_source, guid=guid, dynamic=dynamic)
  File "lib/galaxy/tools/__init__.py", line 1025, in parse
    biotools_entry = self.app.biotools_metadata_source.get_biotools_metadata(biotools_reference)
AttributeError: 'ValidationContext' object has no attribute 'biotools_metadata_source'
```

quast also has a bio.tools reference and failed with the same message.  Tools without bio.tools references are installed normally.

**Galaxy Version and/or server at which you observed the bug**
release_21.09 (ff82ff3)

**To Reproduce**
1. echo -e "galaxy:\n\tadmin_users: admin@a.b" > myconfig.yml; export GALAXY_CONFIG_FILE=myconfig.yml
2. bash run.sh
3. register in localhost galaxy as the admin user
4. install the latest fastqc from the admin tab and watch the logs

**Expected behavior**
fastqc installed

**Screenshots**

**Additional context**

### PR DESCRIPTION
**Pull request by mvdbeek**
Fixes https://github.com/galaxyproject/galaxy/issues/12918:
```
galaxy.tool_shed.tools.tool_validator ERROR 2021-11-15 15:55:52,992 [pN:main.web.1,p:5536,w:1,m:0,tN:uWSGIWorker1Core3] Caught exception loading tool from /Users/cat/dev/galaxy/database/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/3d0c7bdf12f5/fastqc/rgFastQC.xml:
Traceback (most recent call last):
  File "lib/galaxy/tool_shed/tools/tool_validator.py", line 80, in load_tool_from_config
    tool = create_tool_from_source(config_file=full_path, app=self.app, tool_source=tool_source, repository_id=repository_id, allow_code_files=False)
  File "lib/galaxy/tools/__init__.py", line 266, in create_tool_from_source
    tool = ToolClass(config_file, tool_source, app, **kwds)
  File "lib/galaxy/tools/__init__.py", line 585, in __init__
    raise e
  File "lib/galaxy/tools/__init__.py", line 582, in __init__
    self.parse(tool_source, guid=guid, dynamic=dynamic)
  File "lib/galaxy/tools/__init__.py", line 1025, in parse
    biotools_entry = self.app.biotools_metadata_source.get_biotools_metadata(biotools_reference)
AttributeError: 'ValidationContext' object has no attribute 'biotools_metadata_source'
```

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

### CODE REVIEW

File: lib/galaxy/tools/repositories.py [hunk 1]
```diff
@@ -56,15 +61,18 @@ def from_app(app, work_dir=None):
...
+            with ValidationContext(app_name=app.name,
+                                   security=app.security,
+                                   model=app.model,
+                                   tool_data_path=work_dir,
+                                   shed_tool_data_path=work_dir,
+                                   tool_data_tables=tool_data_tables,
+                                   registry=app.datatypes_registry,
+                                   hgweb_config_manager=getattr(app, 'hgweb_config_manager', None),
+                                   biotools_metadata_source=getattr(app, 'biotools_metadata_source', None),
+                                   ) as app:
```
Comment (nsoranzo) [inline lib/galaxy/tools/repositories.py]:
```suggestion
            with ValidationContext(
                app_name=app.name,
                security=app.security,
                model=app.model,
                tool_data_path=work_dir,
                shed_tool_data_path=work_dir,
                tool_data_tables=tool_data_tables,
                registry=app.datatypes_registry,
                hgweb_config_manager=getattr(app, 'hgweb_config_manager', None),
                biotools_metadata_source=getattr(app, 'biotools_metadata_source', None),
            ) as app:
```

**Review by nsoranzo — APPROVED** [review]

**Comment by nsoranzo** [comment]
Did you mean to target the release_21.09 branch?

**Comment by mvdbeek** [comment]
Yep, thanks for catching that!

**Comment by jmchilton** [comment]
The integration test failures seem unrelated. Are there problems with the node configuration in 21.09 CI?

**Review by jmchilton — APPROVED** [review]

**Comment by mvdbeek** [comment]
I had seen that a few times over the last weeks ... but I think I also didn't base this on the newest commit, and this seems to be specific to circleci, we do also build the client on multiple tests in the github workflows.

**Pull request merged**